src/hotspot/cpu/aarch64/stubGenerator_aarch64.cpp

```diff
@@ -4660,19 +4659,19 @@ class StubGenerator: public StubCodeGenerator {
     const Register y     = r2;
     const Register ylen  = r3;
     const Register z     = r4;
-    const Register zlen  = r5;
 
-    const Register tmp1  = r10;
-    const Register tmp2  = r11;
-    const Register tmp3  = r12;
-    const Register tmp4  = r13;
-    const Register tmp5  = r14;
-    const Register tmp6  = r15;
-    const Register tmp7  = r16;
+    const Register tmp0  = r10;
+    const Register tmp1  = r11;
+    const Register tmp2  = r12;
+    const Register tmp3  = r13;
+    const Register tmp4  = r14;
+    const Register tmp5  = r15;
+    const Register tmp6  = r16;
+    const Register tmp7  = r17;
```

Comment:
Why not minimize changes and continue to use r5 for tmp0?  I see no need for r17 or to reassign all the other tmp registers.